core/src/main/java/com/scalar/db/common/ActiveTransactionManagedDistributedTransactionManager.java

```diff
@@ -40,14 +53,14 @@ public ActiveTransactionManagedDistributedTransactionManager(DatabaseConfig conf
             TRANSACTION_EXPIRATION_INTERVAL_MILLIS,
             (id, t) -> {
               logger.warn("The transaction is expired. transaction ID: {}", id);
-              try {
-                t.rollback();
-              } catch (Exception e) {
-                logger.warn("Rollback failed. transaction ID: {}", id, e);
-              }
+              transactionExpirationHandler.get().accept(id, t);
             });
   }
 
+  public void setTransactionExpirationHandler(BiConsumer<String, DistributedTransaction> handler) {
```

Comment:
We will be able to set a transaction expiration handler after this change.

Reply:
@brfrn169 Can you tell me how the custom handler would be like?

Reply:
I'm going to use it for the ScalarDB Auth thing. Theoretically, we can do anything for expired transactions. For example, we can output detail logs for the expired transactions.